debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/CacheMemorySource.java

```diff
@@ -0,0 +1,166 @@
+ * General Public License, version 2 with the GNU Classpath                                                                                                                  
+ * Exception [1] and GNU General Public License, version 2 with the                                                                                                          
+ * OpenJDK Assembly Exception [2].                                                                                                                                           
+ *                                                                                                                                                                           
+ * [1] https://www.gnu.org/software/classpath/license.html                                                                                                                   
+ * [2] http://openjdk.java.net/legal/assembly-exception.html                                                                                                                 
+ *                                                                                                                                                                           
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception                                       
+ *******************************************************************************/
+package com.ibm.j9ddr.tools.ddrinteractive;
+
+import java.nio.ByteOrder;
+import java.nio.ByteBuffer;
+import java.nio.Buffer;
+import java.lang.reflect.Field;
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.corereaders.Platform;
+import com.ibm.j9ddr.corereaders.memory.AbstractMemory;
+import com.ibm.j9ddr.corereaders.memory.IMemorySource;
+import com.ibm.j9ddr.corereaders.memory.IMemoryRange;
+import com.ibm.j9ddr.corereaders.memory.ProtectedMemoryRange;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import sun.nio.ch.DirectBuffer;
+
+public class CacheMemorySource extends ProtectedMemoryRange implements IMemorySource
+        {
+
+	    private ByteBuffer bb;
+	    protected final int asid;
+	    protected final long baseAddress;
+	    protected final long size;	    
+
+	    public CacheMemorySource(long address,
+				int size)
+	    {
+
+		super(address,(long)size);
+		this.asid = 0;
+		this.baseAddress = address;
+                this.size = (long)size;
+		this.bb = setup(address, size);
+                }
+
+	    /*
+	     * Since the src is direct bb backed
+	     * it makes most sense to expose the following accessor
+	     * methods from the src implementation
+	     * as opposed to a getBytes (rewrapped) result 
+	     */
+
+	    public int getUnsignedShort(long address) {
+		int index = (int) (address - this.baseAddress);
+		if(bb.order() == ByteOrder.BIG_ENDIAN){
```

Comment:
Formatting:
* space after keywords (e.g. `if`)
* space before `{` and after `}`
* space both sides of binary operators (e.g. `+`)